include/inexor/vulkan-renderer/wrapper/command_buffer.hpp

```diff
@@ -33,59 +77,255 @@ class CommandBuffer {
     CommandBuffer &operator=(const CommandBuffer &) = delete;
     CommandBuffer &operator=(CommandBuffer &&) = delete;
 
-    /// @brief Call vkBeginCommandBuffer.
-    /// @note Sometimes it's useful to pass VK_COMMAND_BUFFER_USAGE_SIMULTANEOUS_USE_BIT to specify that a command
-    /// buffer can be resubmitted to a queue while it is in the pending state, and recorded into multiple primary
-    /// command buffers. Otherwise, synchronization must be done using a VkFence.
-    /// @param flags The command buffer usage flags, 0 by default.
-    void begin(VkCommandBufferUsageFlags flags = 0) const;
+    /// Call vkBeginCommandBuffer
+    /// @param flags The command buffer usage flags, ``VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT`` by default
+    const CommandBuffer & // NOLINT
+    begin_command_buffer(VkCommandBufferUsageFlags flags = VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT) const;
```

Comment:
`begin_command_buffer` and `end_command_buffer` can be removed and made automatic

Reply:
I made them private. Done.